### PR TITLE
[Create login endpoint] & [Tests over new login endpoint] & [Create tests over UserViewSet]

### PR DESCRIPTION
**Pull request by Dasher83**
A login view added from the Django Rest Framework. A customized version was exposed in the API app.
Tests were added over this customized login endpoint.
Tests were added over the UserViewSet.

### CODE REVIEW

File: mysite/api/serializers.py [hunk 1]
```diff
@@ -38,10 +38,23 @@ class Meta:
...
 class UserUpdateSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
         exclude = ("id", "date_joined", "last_login")
+
+    def update(self, instance, validated_data):
+        super().update(instance, validated_data)
+        instance.set_password(instance.password)
+        instance.save()
+        return instance
```
Comment (joaquinco) [inline mysite/api/serializers.py:60]: You might want to support partial updates too

File: mysite/api/tests.py [hunk 2]
```diff
@@ -816,3 +817,484 @@ def test_get_one_user_non_existent(self):
...
+    def test_post_user_successfully(self):
+        """
+        Create a new user without errors
+        """
+        user_data = {
+            "username": "test",
+            "email": "test@gmail.com",
+            "password": "1234",
+            "first_name": "Mr. Test",
+            "last_name": "Unit",
```
Comment (joaquinco) [inline mysite/api/tests.py:832]: Maybe this can help you creating fake realistic data https://github.com/joke2k/faker

**Review by joaquinco — APPROVED** [review]

File: mysite/api/tests.py (hunk 2 above)
Comment (Dasher83) [reply mysite/api/tests.py:832]: That would be nice. Do you think I should add it in the features strictly related to training? If not, I will add it to a list of features to add for fun after training to make the project better. Thanks!

File: mysite/api/serializers.py (hunk 1 above)
Comment (Dasher83) [reply mysite/api/serializers.py:60]: Yes, a patch could also be supported for the API, I'll create a PR for it a little down the road. Thanks!

File: mysite/api/tests.py (hunk 2 above)
Comment (joaquinco) [reply mysite/api/tests.py:832]: I leave that up to you

**Review by dcousin24 — APPROVED** [review]

**Pull request merged**